storage/graphdriver/rawblock/driver.go

```diff
@@ -0,0 +1,245 @@
+	return [][2]string{
+		{"Backing Filesystem", d.backingFs},
+		{"Support Copy-On-Write", fmt.Sprintf("%v", d.cow)},
+		{"Block Filesystem", d.blockFs},
+		{"Block Size", fmt.Sprintf("%dGB", d.blockSize)},
+	}
+}
+
+// GetMetadata is used for implementing the graphdriver.ProtoDriver interface.
+func (d *Driver) GetMetadata(id string) (map[string]string, error) {
+	return nil, nil
+}
+
+// Cleanup is used to implement graphdriver.ProtoDriver. There is no cleanup required for this driver.
+func (d *Driver) Cleanup() error {
+	return nil
+}
+
+func (d *Driver) createBaseBlock(id, mountLabel string) error {
+	//opts := []string{"level:s0"}
+	//if _, mountLabel, err := label.InitLabels(opts); err == nil {
+	//	label.SetFileLabel(dir, mountLabel)
+	//}
+	if out, err := exec.Command("truncate", fmt.Sprintf("--size=%dG", d.blockSize), d.block(id)).CombinedOutput(); err != nil {
+		return fmt.Errorf("Failed to create block:%v:%s", err, string(out))
+	}
+	if out, err := exec.Command("losetup", "-f", "--show", d.block(id)).CombinedOutput(); err != nil {
+		return fmt.Errorf("Failed to lopsetup the block:%v:%s", err, string(out))
+	} else {
+		defer exec.Command("losetup", "-d", string(out)).CombinedOutput()
+		if out1, err := exec.Command("mkfs", "-t", d.blockFs, d.block(id)).CombinedOutput(); err != nil {
+			return fmt.Errorf("Failed to mkfs the block:%v:%s", err, string(out1))
+		}
+		return nil
+	}
+
+}
+
+// Create prepares the filesystem for the rawblock driver and copies the block from the parent.
+func (d *Driver) Create(id, parent, mountLabel string) error {
+	rootUID, rootGID, err := idtools.GetRootUIDGID(d.uidMaps, d.gidMaps)
+	if err != nil {
+		return err
+	}
+	if err := idtools.MkdirAllAs(filepath.Dir(d.block(id)), 0700, rootUID, rootGID); err != nil {
+		return err
+	}
+	if err := idtools.MkdirAllAs(d.mnt(id), 0755, rootUID, rootGID); err != nil {
+		return err
+	}
+	if parent == "" {
+		return d.createBaseBlock(id, mountLabel)
+	}
+	// TODO: use syscall directly, cp --reflink supports btrfs only currently
+	if out, err := exec.Command("cp", "-a", "--reflink=auto", d.block(parent), d.block(id)).CombinedOutput(); err != nil {
+		return fmt.Errorf("Failed to reflink:%v:%s", err, string(out))
+	}
+	return nil
+}
+
+func (d *Driver) block(id string) string {
+	return filepath.Join(d.home, "blocks", id)
+}
+
+func (d *Driver) mnt(id string) string {
+	return filepath.Join(d.home, "mnt", id)
+}
+
+func joinMountOptions(a, b string) string {
+	if a == "" {
+		return b
+	}
+	if b == "" {
+		return a
+	}
+	return a + "," + b
+}
+
+func (d *Driver) mount(id string, mountLabel string) error {
+	options := "loop"
+
+	if d.blockFs == "xfs" {
+		// XFS needs nouuid or it can't mount filesystems with the same fs
+		options = joinMountOptions(options, "nouuid")
+	}
+
+	options = joinMountOptions(options, label.FormatMountLabel("", mountLabel))
+
+	//if err := mount.Mount(d.block(id), d.mnt(id), d.blockFs, options); err != nil {
```

Comment:
we need to add loopback related code when use mount.Mount(). using mount commandline is much simpler. The 'Put()' function forgot to detach the loopdev. it needs to be fixed.